packages/apollo-cache-control/src/index.ts

```diff
@@ -141,14 +141,21 @@ export class CacheControlExtension<TContext = any>
     if (this.options.calculateHttpHeaders && o.graphqlResponse.http) {
       const overallCachePolicy = this.computeOverallCachePolicy();
 
-      if (overallCachePolicy) {
-        o.graphqlResponse.http.headers.set(
-          'Cache-Control',
-          `max-age=${
-            overallCachePolicy.maxAge
-          }, ${overallCachePolicy.scope.toLowerCase()}`,
-        );
+      if (
+        o.graphqlResponse.errors ||
+        !o.graphqlResponse.data ||
```

Comment:
Hmm, I didn't catch or mention this the first time, but wouldn't checking for `o.graphqlResponse.errors` (and not checking for `o.graphqlResponse.data`) be sufficient?  I'm curious what the motivation for adding that extra line was.

Ref: https://graphql.github.io/graphql-spec/June2018/#sec-Response-Format

Reply:
Yes, you are right. At first I also implemented it like this. But then I had a look at the "apollo-server-plugin-response-cache" package, from which I grabbed this snippet:
https://github.com/apollographql/apollo-server/blob/7db109efed2a193e0b459030ccd748d1f329fc82/packages/apollo-server-plugin-response-cache/src/ApolloServerPluginResponseCache.ts#L243-L261

So my motivation for adding this basically was to be aligned with the "apollo-server-plugin-response-cache" package, because the logic is kind of similar. A response should only be cached if it does not include any errors and if it contains data. This should be the case for Memory/Redis caching ("apollo-server-plugin-response-cache") as well as for CDN or browser caching ("apollo-cache-control").

The `overallCachePolicy.maxAge <= 0` part is also kind of unnecessary in this case, since the `computeOverallCachePolicy` method already returns `undefined` if `lowestMaxAge` is `0`.

I could change the implementation to this, if you think that it makes more sense:

```ts
public willSendResponse?(o: { graphqlResponse: GraphQLResponse }) {
  if (this.options.calculateHttpHeaders && o.graphqlResponse.http) {
    if (o.graphqlResponse.errors) {
      return;
    }

    const overallCachePolicy = this.computeOverallCachePolicy();

    if (overallCachePolicy) {
      o.graphqlResponse.http.headers.set(
        'Cache-Control',
        `max-age=${
          overallCachePolicy.maxAge
        }, ${overallCachePolicy.scope.toLowerCase()}`,
      );
    }
  }
}
```

This way the `computeOverallCachePolicy` is not even called if the response contains errors.

Reply:
Good point! I appreciate you trying to align it, but I think I like this better.  Thank you for explaining it so clearly, as well!

I tend to favor `return`ing early (and reducing indentation) when possible, so, thoughts on the first line being?:

```js
if (!this.options.calculateHttpHeaders || !o.graphqlResponse.http || o.graphqlResponse.errors) {
  return;
}
```

If you agree, feel free to adjust and I'd approve! (Also, if you don't agree with this bit, still feel free to adjust for your previous suggestion and I'll still approve.) 😃

Reply:
Just changed the implementation to your suggested solution and rebased the branch from master.